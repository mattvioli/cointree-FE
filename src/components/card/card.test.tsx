import { render, screen } from "@testing-library/react";
import { Card } from "./index";

describe("Card component", () => {
  it("renders children correctly", () => {
    const childText = "This is the content inside the card.";
    render(<Card>{childText}</Card>);

    const cardElement = screen.getByText(childText);

    expect(cardElement).toBeInTheDocument();
  });

  it("has the correct Tailwind classes applied", () => {
    const childText = "This is the content inside the card.";
    render(<Card>{childText}</Card>);

    const cardElement = screen.getByText(childText);

    expect(cardElement).toMatchSnapshot();
  });
});
