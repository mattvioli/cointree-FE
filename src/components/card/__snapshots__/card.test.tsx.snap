// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component has the correct Tailwind classes applied 1`] = `
<div
  class="flex flex-col gap-4 mt-8 shadow-lg rounded-2xl p-6 bg-white items-center"
>
  This is the content inside the card.
</div>
`;
